src/internal/packager2/layout/create.go

```diff
@@ -0,0 +1,467 @@
+			}
+
+			// Make sure dst reflects the actual file or directory.
+			updatedExtractedFileOrDir := filepath.Join(destinationDir, file.ExtractPath)
+			if updatedExtractedFileOrDir != dst {
+				if err := os.Rename(updatedExtractedFileOrDir, dst); err != nil {
+					return fmt.Errorf(lang.ErrWritingFile, dst, err)
+				}
+			}
+		} else {
+			if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, file.Source), dst); err != nil {
+				return fmt.Errorf("unable to copy file %s: %w", file.Source, err)
+			}
+		}
+
+		// Change the source to the new relative source directory (any remote files will have been skipped above)
+		component.Files[filesIdx].Source = rel
+
+		// Remove the extractPath from a skeleton since it will already extract it
+		component.Files[filesIdx].ExtractPath = ""
+
+		// Abort packaging on invalid shasum (if one is specified).
+		if file.Shasum != "" {
+			if err := helpers.SHAsMatch(dst, file.Shasum); err != nil {
+				return err
+			}
+		}
+
+		if file.Executable || helpers.IsDir(dst) {
+			err = os.Chmod(dst, helpers.ReadWriteExecuteUser)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = os.Chmod(dst, helpers.ReadWriteUser)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for dataIdx, data := range component.DataInjections {
+		rel := filepath.Join(string(DataComponentDir), strconv.Itoa(dataIdx), filepath.Base(data.Target.Path))
+		dst := filepath.Join(compBuildPath, rel)
+
+		if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, data.Source), dst); err != nil {
+			return fmt.Errorf("unable to copy data injection %s: %s", data.Source, err.Error())
+		}
+
+		component.DataInjections[dataIdx].Source = rel
+	}
+
+	// Iterate over all manifests.
+	for manifestIdx, manifest := range component.Manifests {
+		for fileIdx, path := range manifest.Files {
+			rel := filepath.Join(string(ManifestsComponentDir), fmt.Sprintf("%s-%d.yaml", manifest.Name, fileIdx))
+			dst := filepath.Join(compBuildPath, rel)
+
+			// Copy manifests without any processing.
+			if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, path), dst); err != nil {
+				return fmt.Errorf("unable to copy manifest %s: %w", path, err)
+			}
+
+			component.Manifests[manifestIdx].Files[fileIdx] = rel
+		}
+
+		for kustomizeIdx, path := range manifest.Kustomizations {
+			// Generate manifests from kustomizations and place in the package.
+			kname := fmt.Sprintf("kustomization-%s-%d.yaml", manifest.Name, kustomizeIdx)
+			rel := filepath.Join(string(ManifestsComponentDir), kname)
+			dst := filepath.Join(compBuildPath, rel)
+
+			if err := kustomize.Build(filepath.Join(packagePath, path), dst, manifest.KustomizeAllowAnyDirectory); err != nil {
+				return fmt.Errorf("unable to build kustomization %s: %w", path, err)
+			}
+		}
+
+		// remove kustomizations
+		component.Manifests[manifestIdx].Kustomizations = nil
+	}
+
+	// Write the tar component.
+	size, err := helpers.GetDirSize(compBuildPath)
```

Comment:
Confused on why this is occurring, and can this just be replaced with an `os.Stat`?

Reply:
This could be replaced by `os.ReadDir` as we dont really care about the size, just that the directory is not empty.